tests/webapp/api/test_artifact_api.py

```diff
@@ -68,3 +73,52 @@ def test_artifact_detail_bad_project(webapp, jm):
     assert resp.json == {"detail": "No project with name foo"}
 
     jm.disconnect()
+
+
+def test_artifact_create_text_log_summary(webapp, eleven_jobs_processed,
+                                          mock_send_request, monkeypatch,
+                                          sample_data, jm):
+    """
+    test creating a single artifact with bug suggestions
+    """
+    from treeherder.model import bug_suggestions
+
+    def _get_bug_suggestions(params):
+        return ["foo", "bar"]
+
+    monkeypatch.setattr(bug_suggestions, "get_bug_suggestions", _get_bug_suggestions)
+
+    credentials = OAuthCredentials.get_credentials(jm.project)
+
+    job = jm.get_job_list(0, 1)[0]
+    tls = sample_data.text_log_summary
+
+    tac = thclient.TreeherderArtifactCollection()
+    ta = thclient.TreeherderArtifact({
+        'type': 'json',
+        'name': 'text_log_summary',
+        'blob': json.dumps(tls['blob']),
+        'job_guid': job['job_guid']
+    })
+    tac.add(ta)
+
+    req = thclient.TreeherderRequest(
+        protocol='http',
+        host='treeherder.mozilla.org',
+        project=jm.project,
+        oauth_key=credentials['consumer_key'],
+        oauth_secret=credentials['consumer_secret']
+        )
+
+    # Post the request to treeherder
+    resp = req.post(tac)
+    assert resp.status_int == 200
+    assert resp.body == '{"message": "Artifacts stored successfully"}'
+
+    with ArtifactsModel(jm.project) as artifacts_model:
+        artifacts = artifacts_model.get_job_artifact_references(job["id"])
+
+    artifact_names = [x['name'] for x in artifacts]
+    assert set(artifact_names) == set(['Bug suggestions', 'text_log_summary'])
+
```

Comment:
Should we not also test that the bug suggestions ('foo', 'bar') were stored correctly here?